### PR TITLE
Orphan facts. Closes 828.

### PR DESCRIPTION
**Pull request by mmcdermott**

### CODE REVIEW

File: src/components/organizations/show/facts/factGraph.js [hunk 1]
```diff
@@ -140,7 +142,13 @@ export class FactGraph extends Component {
       edges.push(...parents.map(p => ({input: locationById(p), output: locationById(id), pathStatus})))
     })
 
-    return { items, edges: edges.filter(e => !!e.input && !!e.output) }
+    const bad_edges = _.remove(edges, ({input, output}) => !_utils.allPresent(input.row, input.column, output.row, output.column))
+    if (!_.isEmpty(bad_edges)) {
+      console.warn(bad_edges.legnth, 'BAD EDGES ENCOUNTERED!')
```
Comment (OAGr) [inline src/components/organizations/show/facts/factGraph.js]: length*

File: src/components/organizations/show/facts/factGraph.js [hunk 2]
```diff
@@ -72,6 +72,8 @@ export class FactGraph extends Component {
       component: <SpaceCard space={s} />,
     }))
 
+    const orphanedFactNodes = _.remove(factNodes, _.negate(allParentsWithin(spaceNodes)))
```
Comment (OAGr) [inline src/components/organizations/show/facts/factGraph.js]: leave comment to explain this.

File: src/modules/facts/reducer.js [hunk 3]
```diff
@@ -108,6 +108,23 @@ export function factsR(state = INITIAL_STATE, action) {
         ],
       }
     }
+    case 'SPACES_DELETE_SUCCESS': {
```
Comment (OAGr) [inline src/modules/facts/reducer.js]: why is this necessary for this PR?

File: src/modules/metrics/actions.js [hunk 4]
```diff
@@ -22,11 +21,23 @@ export function addMetric(item) {
   }
 }
 
-//spaceId must be done before the metric is removed here.
 export function removeMetrics(ids) {
```
Comment (OAGr) [inline src/modules/metrics/actions.js]: any way this could happen on the server?  I realize that would be tricky but it does seem more right to me.

File: src/modules/metrics/actions.js [hunk 5]
```diff
@@ -22,11 +21,23 @@ export function addMetric(item) {
...
+
     const spaceId = findSpaceId(getState, ids[0])
+
+    const {organizations, spaces, facts: {organizationFacts}} = getState()
+    const organizationId = e.collections.gget(spaces, spaceId, 'id', 'organization_id')
+    if (!!organizationId) {
+      const organization = e.collections.get(organizations, organizationId)
+      const facts = e.organization.findFacts(organizationId, organizationFacts)
+      const factsToDelete = _.filter(facts, f => f.exported_from_id === spaceId && ids.includes(f.metric_id))
+      factsToDelete.forEach(fact => {
```
Comment (OAGr) [inline src/modules/metrics/actions.js]: It would be nicer if they could be deleted as part of one action, though i realize that practically speaking one metric removal should only go to one fact removal.

Comment (mmcdermott) [inline src/modules/metrics/actions.js]: I agree, but soon I'd like to refactor the facts client side storage (get rid of the organizationFacts middle layer and just store facts by ID like everything else) and this section will undergo heavy changes then that will make it more amenable to such a change, so I'd prefer to make that change then.

File: src/modules/metrics/actions.js (hunk 4 above)
Comment (mmcdermott) [inline src/modules/metrics/actions.js]: We talked offline, decided to leave this here for now.
